core/tests/protractor_utils/AdminPage.js

```diff
@@ -292,6 +333,154 @@ var AdminPage = function() {
     }
   };
 
+  this.getMiscTab = async function() {
+    await action.click('Misc Tab Button', miscTabButton);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.uploadTopicSimilarities =
+    async function(relativePathToSimilaritiesFile, isValidFile) {
+      var absPath = path.resolve(__dirname, relativePathToSimilaritiesFile);
+      await waitFor.visibilityOf(chooseSimilarityFileInput,
+        'Similarity upload form is not visible');
+      await chooseSimilarityFileInput.sendKeys(absPath);
+      await action.click('Upload button', similarityFileUploadButton);
+      if (isValidFile) {
+        var text = 'Topic similarities uploaded successfully.';
```

Comment:
Don't use if else in this function. The test block should have the code for expected behavior. Maybe follow this structure:

uploadTopicSimilarities(correctFilename)
expectSuccesMessage()

uploadTopicSimilarities(incorrectFilename)
expectFailureMessage()